main/src/library/Fixpoint/IndexSelection.flix

```diff
@@ -98,72 +98,72 @@ mod Fixpoint {
                     case _ => exp
                 }, notGround) |>
                 // Partition into those equalities that have `var` on the lhs and those that don't.
-                List.partition(exp -> match exp {
+                Vector.partition(exp -> match exp {
                     case Eq(RowLoad(row1, _), RowLoad(row2, _)) => row1 != row2 and row1 == var
                     case Eq(RowLoad(row, _), Lit(_)) => row == var
                     case _ => false
                 });
             let (prefixQuery, rest2) = longestPrefixQuery(varQuery);
-            let test = rest1 ::: rest2;
-            if (List.isEmpty(prefixQuery))
-                if (List.isEmpty(test))
+            let test = Vector.append(rest1, rest2);
+            if (Vector.isEmpty(prefixQuery))
+                if (Vector.isEmpty(test))
                     let search = RelOp.Search(var, ramSym, innerOp);
                     (search, ground)
                 else
                     let search = RelOp.Search(var, ramSym, RelOp.If(test, innerOp));
                     (search, ground)
             else
                 let query =
-                    List.map(x -> match x {
+                    Vector.map(x -> match x {
                         case Eq(RamTerm.RowLoad(_, j), rhs) => (j, rhs)
                         case _ => ???
                     }, prefixQuery);
-                if (List.isEmpty(test))
+                if (Vector.isEmpty(test))
                     let search = RelOp.Query(var, ramSym, query, innerOp);
                     (search, ground)
                 else
                     let search = RelOp.Query(var, ramSym, query, RelOp.If(test, innerOp));
                     (search, ground)
-        case RelOp.Query(_) => (op, Nil)
-        case RelOp.Project(_) => (op, Nil)
+        case RelOp.Query(_) => (op, Vector.empty())
+        case RelOp.Project(_) => (op, Vector.empty())
         case RelOp.If(test, then) =>
             let (innerOp, innerGround) = queryOp(then, freeVars);
-            (innerOp, test ::: innerGround)
-        case RelOp.Loop(_) => ??? // TODO
+            (innerOp, Vector.append(test, innerGround))
+        case RelOp.Loop(_) => ??? // TODO;
     }
 
-    def longestPrefixQuery(varQuery: List[BoolExp[v]]): (List[BoolExp[v]], List[BoolExp[v]]) =
-        use Fixpoint.Ram.BoolExp.Eq;
-        use Fixpoint.Ram.RamTerm.{RowLoad, Lit};
+    def longestPrefixQuery(varQuery: Vector[BoolExp[v]]): (Vector[BoolExp[v]], Vector[BoolExp[v]]) =
```

Comment:
Perhaps a helper function like this (and others should be left to work on lists).